.nvidia-ci.yml

```diff
@@ -127,3 +127,67 @@ release:ngc-ubi8:
   extends:
     - .release:ngc
     - .dist-ubi8
+
+
+# Define the external image signing steps for NGC
+# Download the ngc cli binary for use in the sign steps
+.ngccli-setup:
+  before_script:
+    - apt-get update && apt-get install -y curl unzip jq
+    - |
+      export NGCLI_VERSION="3.41.4"
+      NGC_VERSION_URL="https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions"
+      json_data=$(curl -s $NGC_VERSION_URL)
+      # Extract the latest version from the JSON data using jq
+      latest_version=$(echo $json_data | jq -r '.recipe.latestVersionIdStr')
+      echo "${latest_version}"
+      if [ -n "$latest_version" ]; then
+        export NGCLI_VERSION=$latest_version
```

Comment:
@elezar latest version will override default version . if json disassembly  fails ,ngc default version will be taken.

Reply:
Taking a step back, is there a need to always upgrade to the latest version? Is it better to pin / stick to a NGCCLI_VERSION that we have tested and know that works in our CI?

Reply:
> Taking a step back, is there a need to always upgrade to the latest version? Is it better to pin / stick to a NGCCLI_VERSION that we have tested and know that works in our CI?

upgrade to latest version is not needed .  
with the current code , we always download latest version instead of upgrading it to latest . 
The default version will be 3.41.4 in case JSON disassembly fails.

Reply:
> Line: 138
In that case, shall we remove export NGCLI_VERSION="3.41.4", as it's not needed?

Yes, that is what Evan is suggesting. We can remove this line.

With Evan's suggestion, we always determine the latest version of the NGC CLI and use it, unless NGCLI_VERSION is set already (e.g. we manually set the variable when invoking the pipeline or set this variable at the project / org level).